analyzer/QUIC.spicy

```diff
@@ -379,19 +380,15 @@ type Packet = unit(from_client: bool, context: ConnectionIDInfo&) {
       context.did_ssl_begin = True;
       }
 @endif
-  }
 
-  # Peek into the header to check if it's a SHORT or LONG header
-  # and get the LONG packet type.
-  : InitialByte &try {
-    self.header_form = $$.initialbyte.header_form;
-    self.long_packet_type = $$.initialbyte.long_packet_type;
+    self.start = self.input();
   }
 
-  # Capture all the packet bytes if we're still have a chance of decrypting the INITIAL PACKETS
-  fpack: AllData &try if ( self.header_form == HeaderForm::LONG &&
-                           self.long_packet_type == LongPacketType::INITIAL &&
-                           should_buffer(context, from_client) );
+  # Peek into the first byte and determine the header type.
+  : uint8 {
+    self.header_form = ($$ & 0x80 == 0x80) ? HeaderForm::LONG : HeaderForm::SHORT;
```

Comment:
Partially out of curiosity, does using bitwise operations make a noticeable perf difference as opposed to an anonymous `bitfield`? If it didn't I'd argue for a (possibly stripped down) `bitfield` since it is just more readable.

Reply:
Reverted to the bitfield variant. Also liked your idea of bit based lookahead parsing, but deferred for another time.